circleci: failed tests: TestReplicaRefreshPendingCommandsTicks

The following test appears to have failed:

[#21035](https://circleci.com/gh/cockroachdb/cockroach/21035):

```
I160805 17:01:10.326632 storage/replica.go:2662  gossiping sentinel from store 1, range 1
I160805 17:01:10.327019 storage/replica.go:2669  gossiping first range from store 1, range 1: [{1 1 1}]
I160805 17:01:10.332604 storage/replica.go:2662  gossiping sentinel from store 1, range 1
I160805 17:01:10.332937 storage/replica.go:2669  gossiping first range from store 1, range 1: [{1 1 1}]
I160805 17:01:10.337015 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- FAIL: TestReplicaRefreshPendingCommandsTicks (0.03s)
	replica_test.go:6111: 2: expected 1 reproprosed commands, but found []
=== RUN   TestReplicaDoubleRefurbish
W160805 17:01:10.353278 gossip/gossip.go:1015  not connected to cluster; use --join to specify a connected node
I160805 17:01:10.354520 storage/engine/rocksdb.go:353  opening in memory rocksdb instance
I160805 17:01:10.362507 storage/replica.go:2640  gossiping cluster id {"beedb931-6964-4a1e-bd54-4ffc248f0d79"} from store 1, range 1
I160805 17:01:10.364305 storage/replica.go:2027  store=1:1 range=1 [/Min-/Max): applying command with forced error: storage/replica.go:1952: no-op on empty Raft entry
E160805 17:01:10.365065 storage/replica.go:2062  store=1:1 range=1 [/Min-/Max): error executing raft command: storage/replica.go:1952: no-op on empty Raft entry
I160805 17:01:10.366142 storage/replica_trigger.go:308  store=1:1 range=1 [/Min-/Max): new range lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 900ms following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0s [physicalTime=1970-01-01 00:00:00 +0000 UTC]
I160805 17:01:10.366206 storage/replica.go:2662  gossiping sentinel from store 1, range 1
I160805 17:01:10.366445 storage/replica.go:2669  gossiping first range from store 1, range 1: [{1 1 1}]
--
I160805 17:02:16.224497 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- PASS: TestEagerReplication (0.28s)
=== RUN   Example_rebalancing
W160805 17:02:16.280178 gossip/gossip.go:1015  not connected to cluster; use --join to specify a connected node
--- PASS: Example_rebalancing (1.08s)
FAIL
FAIL	github.com/cockroachdb/cockroach/storage	80.296s
=== RUN   TestBatchBasics
I160805 17:02:27.481602 storage/engine/rocksdb.go:353  opening in memory rocksdb instance
I160805 17:02:27.485245 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.02s)
=== RUN   TestBatchRepr
I160805 17:02:27.496747 storage/engine/rocksdb.go:353  opening in memory rocksdb instance
I160805 17:02:27.498398 storage/engine/rocksdb.go:398  closing in-memory rocksdb instance
--- PASS: TestBatchRepr (0.02s)
=== RUN   TestBatchGet
I160805 17:02:27.520175 storage/engine/rocksdb.go:353  opening in memory rocksdb instance

```
Please assign, take a look and update the issue accordingly.

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -6054,26 +6054,38 @@ func TestReplicaRefreshPendingCommandsTicks(t *testing.T) {
 	tc.Start(t)
 	defer tc.Stop()
 
-	r := tc.rng
-	repDesc, err := r.GetReplicaDescriptor()
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if pErr := r.redirectOnOrAcquireLease(context.Background()); pErr != nil {
-		t.Fatal(pErr)
-	}
-
 	// Grab processRaftMu in order to block normal raft replica processing. This
 	// test is ticking the replicas manually and doesn't want the store to be
 	// doing so concurrently.
 	tc.store.processRaftMu.Lock()
 	defer tc.store.processRaftMu.Unlock()
 
+	r := tc.rng
+	repDesc, err := r.GetReplicaDescriptor()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	electionTicks := tc.store.ctx.RaftElectionTimeoutTicks
+
+	{
+		// The verifications of the reproposal counts below rely on r.mu.ticks
+		// starting with a value of 0 (modulo electionTicks). Move the replica into
+		// that state in case the replica was ticked before we grabbed
+		// processRaftMu.
+		r.mu.Lock()
+		ticks := r.mu.ticks
+		r.mu.Unlock()
+		for ; (ticks % electionTicks) != 0; ticks++ {
+			if err := r.tick(); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+
 	// We tick the replica 2*RaftElectionTimeoutTicks. RaftElectionTimeoutTicks
 	// is special in that it controls how often pending commands are reproposed
 	// or refurbished.
-	electionTicks := tc.store.ctx.RaftElectionTimeoutTicks
 	for i := 0; i < 2*electionTicks; i++ {
 		// Add another pending command on each iteration.
 		r.mu.Lock()

